cupy/core/_routines_indexing.pyx

```diff
@@ -350,18 +355,12 @@ cdef ndarray _simple_getitem(ndarray a, list slice_list):
     return v
 
 
-_nonzero_kernel_1d = ElementwiseKernel(
-    'T src, S index', 'raw S dst',
-    'if (src != 0) dst[index - 1] = i',
-    'nonzero_kernel_1d')
-
-
 _nonzero_kernel = ElementwiseKernel(
     'T src, S index', 'raw S dst',
     '''
     if (src != 0){
         for(int j = 0; j < _ind.ndim; j++){
-            ptrdiff_t ind[] = {j, index - 1};
+            ptrdiff_t ind[] = {index - 1, j};
```

Comment:
Please check the strides of the return values.
```py
>>> import numpy, cupy
>>>
>>> for xp in (numpy, cupy):
...     x = xp.arange(12).reshape(3, 4)
...     print(xp.argwhere(x).strides)
...
(8, 88)
(16, 8)
```

Reply:
Hmm, NumPy seems to use different implementation in `numpy.argwhere` and `numpy.nonzero`.
```py
>>> for xp in (numpy, cupy):
...     x = xp.arange(12).reshape(3, 4)
...     print(xp.nonzero(x)[0].strides)
...
(16,)
(16,)
```
This PR fixes the strides of return value of `numpy.nonzero`, so it is OK to keep the current implementaion. However, we may fix later for strides of `numpy.argwhere`

Reply:
@asi1024 Could you take another look at this contiguity issue since @rushabh-v reverted his change on the `nonzero` kernel? I think this is the reason I saw the performance degradation (https://github.com/cupy/cupy/pull/3135#issuecomment-595891437).

Reply:
btw, for `argwhere`, note that the NumPy output is F-contiguous. I guess this is reason for the stride discrepancy.

Reply:
I just recalled that CuPy does not guarantee returning identical strides as in NumPy (#2717), so there's nothing to be fixed!